create_es_bulk_json.py

```diff
@@ -1,24 +1,31 @@
 # coding=utf-8
-import sys, json, subprocess
+import argparse
+import json
+import subprocess
+import sys
+
 
 # SSH into a remote server and find war files under certain directories
 def get_file_names(instance):
-    path = "*/%s/*/current/dist/*.war" % instance
-    ssh = subprocess.Popen(['ssh', '-l', xe_user, xe_host, 'find', banner_home, '-type l -wholename', path, '-exec readlink {} \;'],
-                           shell=False,
-                           stdout=subprocess.PIPE,
-                           stderr=subprocess.PIPE)
+    path = f"*/{instance}/*/current/dist/*.war"
+    command = [
+        'ssh', '-l', xe_user, xe_host, 'find', banner_home,
+        '-type l -wholename', path, '-exec readlink {} \;'
+    ]
+    ssh = subprocess.Popen(
+        command, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    )
     result = ssh.stdout.readlines()
     # If we didn't get any files, something's wrong
-    if result == []:
+    if not result:
         error = ssh.stderr.readlines()
-        print >> sys.stderr, "ERROR: %s" % error
-        sys.exit(1)
+        sys.exit(f"ERROR: {error}")
+
+    return result
 
-    return result    
 
 # Parse the whole filename into the application name and version.
-def parse_file_names(instance, xe_apps):
+def parse_file_names(instance):
     file_names = get_file_names(instance)
 
     for file in file_names:
```

Comment:
Since this uses python3 now, this loop runs into an error:
```
Traceback (most recent call last):
  File "create_es_bulk_json.py", line 71, in <module>
    write_apps_to_file()
  File "create_es_bulk_json.py", line 51, in write_apps_to_file
    parse_file_names(instance)
  File "create_es_bulk_json.py", line 32, in parse_file_names
    app = file.split('-', 1)[0]
TypeError: a bytes-like object is required, not 'str'
```

I did a quick fix: 
```
diff --git a/create_es_bulk_json.py b/create_es_bulk_json.py
index 31a8fef..16a7ddb 100644
--- a/create_es_bulk_json.py
+++ b/create_es_bulk_json.py
@@ -29,6 +29,7 @@ def parse_file_names(instance):
     file_names = get_file_names(instance)

     for file in file_names:
+        file = file.decode('utf_8')
         app = file.split('-', 1)[0]
         version = file.split('-', 1)[-1]
         version = version.split('.war', 1)[0]
```
Probably a better way to do it, I was just troubleshooting.

Reply:
This https://github.com/osu-mist/get-xe-versions-for-es/blob/d86c8d8b0af6e667ea91e597727230517689dcf7/create_es_bulk_json.py#L19 should address this